### PR TITLE
Add BEIS fuel EFs in tonnes

### PR DESCRIPTION
**Pull request by greenercode**
Support kg-CO2e/t for 2021 BEIS fuel emission factors

**Is this a new addition or an update to an existing factor?**

- [x] New

**What category or categories does it affect?**

Fuel 

**What?**

Support BEIS fuel emission factors in kg-CO2e/t (for the year 2021). 

**Why?**

Facilitating industry usage and solutions for fuel emission factors. Usage might require kg-CO2e/t instead of kg-CO2e/l.

### CODE REVIEW

File: OpenEmissionFactorsDB.csv [hunk 1]
```diff
@@ -1246,3 +1246,34 @@ Transport,Rail Travel,passenger_train-route_type_local-fuel_source_electricity,E
...
+Energy,Fuel,fuel_type_marine_fuel_oil-activity_combustion,Marine fuel oil - combustion,kg-CO2e/t,3159.51,,BEIS,2021,GB,11/17/2021,Emission intensity for marine fuel oil. The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
+Energy,Fuel,fuel_type_coal_industrial-activity_combustion,Coal (industrial) - combustion,kg-CO2e/t,2403.84,,BEIS,2021,GB,11/17/2021,Emission intensity for coal (industrial). The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
```
Comment (patstirling) [inline OpenEmissionFactorsDB.csv:1273]: Example line. To align with guidance/current EFs, this line should read: 
- the `ID` field for this EF should read "fuel_type_coal-fuel_use_industrial"
- the `name` field should read "Coal (industrial")
- the `activity` filed should read "combustion"

File: OpenEmissionFactorsDB.csv [hunk 2]
```diff
@@ -1246,3 +1246,34 @@ Transport,Rail Travel,passenger_train-route_type_local-fuel_source_electricity,E
...
+Energy,Fuel,fuel_type_propane-activity_combustion,Propane - combustion,kg-CO2e/t,2997.55,,BEIS,2021,GB,11/17/2021,Emission intensity for propane. The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
+Energy,Fuel,fuel_type_aviation_spirit-activity_combustion,Aviation spirit - combustion,kg-CO2e/t,3192.76,,BEIS,2021,GB,11/17/2021,Emission intensity for aviation spirit. The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
```
Comment (patstirling) [inline OpenEmissionFactorsDB.csv:1256]: could this be updated to reflect other aviation fuel lines e.g. "fuel_type_avgas" and "Aviation fuel (avgas)"

File: OpenEmissionFactorsDB.csv [hunk 3]
```diff
@@ -1246,3 +1246,34 @@ Transport,Rail Travel,passenger_train-route_type_local-fuel_source_electricity,E
...
+Energy,Fuel,fuel_type_coal_home_produced_electricity_generation-activity_combustion,Coal (electricity generation - home produced coal only) - combustion,kg-CO2e/t,2248.82,,BEIS,2021,GB,11/17/2021,Emission intensity for coal (electricity generation - home produced coal only). The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
+Energy,Fuel,fuel_type_natural_gas_100_per_cent_mineral_blend-activity_combustion,Natural gas (100% mineral blend) - combustion,kg-CO2e/t,2555.28,,BEIS,2021,GB,11/17/2021,Emission intensity for aviation spirit natural gas (100% mineral blend). The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
```
Comment (patstirling) [inline OpenEmissionFactorsDB.csv:1279]: This line seems a bit rogue? Description includes "aviation spirit"; it also has the precise same EF as the "LNG" line (1251)

File: OpenEmissionFactorsDB.csv [hunk 4]
```diff
@@ -1246,3 +1246,34 @@ Transport,Rail Travel,passenger_train-route_type_local-fuel_source_electricity,E
...
+Energy,Fuel,fuel_type_aviation_spirit-activity_combustion,Aviation spirit - combustion,kg-CO2e/t,3192.76,,BEIS,2021,GB,11/17/2021,Emission intensity for aviation spirit. The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
+Energy,Fuel,fuel_type_aviation_turbine-activity_combustion,Aviation turbine fuel - combustion,kg-CO2e/t,3181.43,,BEIS,2021,GB,11/17/2021,Emission intensity for aviation turbine fuel. The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
```
Comment (patstirling) [inline OpenEmissionFactorsDB.csv:1257]: could this be updated to reflect other aviation fuel lines e.g. "fuel_type_avtur" and "Aviation turbine fuel (avtur)"

File: OpenEmissionFactorsDB.csv [hunk 5]
```diff
@@ -1246,3 +1246,34 @@ Transport,Rail Travel,passenger_train-route_type_local-fuel_source_electricity,E
...
+Energy,Fuel,fuel_type_diesel_average_biofuel_blend-activity_combustion,Diesel (average biofuel blend) - combustion,kg-CO2e/t,2969.07,,BEIS,2021,GB,11/17/2021,Emission intensity for diesel (average biofuel blend). The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
+Energy,Fuel,fuel_type_diesel_100_per_cent_mineral-activity_combustion,Diesel (100% mineral diesel) - combustion,kg-CO2e/t,3208.76,,BEIS,2021,GB,11/17/2021,Emission intensity for diesel (100 % mineral blend). The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
```
Comment (patstirling) [inline OpenEmissionFactorsDB.csv:1260]: remove (100% mineral diesel) from the name to be comparable to other mineral diesel fuel EFs (this is just standard diesel). The "100% mineral" component can be left in the description

File: OpenEmissionFactorsDB.csv [hunk 6]
```diff
@@ -1246,3 +1246,34 @@ Transport,Rail Travel,passenger_train-route_type_local-fuel_source_electricity,E
...
+Energy,Fuel,fuel_type_petrol_average_biofuel_blend-activity_combustion,Petrol (average biofuel blend) - combustion,kg-CO2e/t,2947.62,,BEIS,2021,GB,11/17/2021,Emission intensity for petrol (average biofuel blend). The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
+Energy,Fuel,fuel_type_petrol_100_per_cent_mineral-activity_combustion,Petrol (100% mineral petrol) - combustion,kg-CO2e/t,3153.9,,BEIS,2021,GB,11/17/2021,Emission intensity for petrol (100% mineral petrol). The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
```
Comment (patstirling) [inline OpenEmissionFactorsDB.csv:1266]: remove "(100% mineral petrol)" from the name to be comparable to other mineral petrol fuel EFs (this is just standard petrol). The "100% mineral" component can be left in the description

File: OpenEmissionFactorsDB.csv [hunk 7]
```diff
@@ -1246,3 +1246,34 @@ Transport,Rail Travel,passenger_train-route_type_local-fuel_source_electricity,E
...
+Energy,Fuel,fuel_type_processed_fuel_oils_residual-activity_combustion,Processed fuel oils - residual oil - combustion,kg-CO2e/t,3229.2,,BEIS,2021,GB,11/17/2021,Emission intensity for processed fuel oils (residual oil). The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
+Energy,Fuel,fuel_type_processed_fuel_oils_distillate-activity_combustion,Processed fuel oils - distillate oil - combustion,kg-CO2e/t,3230.28,,BEIS,2021,GB,11/17/2021,Emission intensity for processed fuel oils (distillate oil). The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
```
Comment (patstirling) [inline OpenEmissionFactorsDB.csv:1268]: suggest name "Processed fuel oils (distillate oil)" to be consistent with naming convention

File: OpenEmissionFactorsDB.csv [hunk 8]
```diff
@@ -1246,3 +1246,34 @@ Transport,Rail Travel,passenger_train-route_type_local-fuel_source_electricity,E
...
+Energy,Fuel,fuel_type_petrol_100_per_cent_mineral-activity_combustion,Petrol (100% mineral petrol) - combustion,kg-CO2e/t,3153.9,,BEIS,2021,GB,11/17/2021,Emission intensity for petrol (100% mineral petrol). The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
+Energy,Fuel,fuel_type_processed_fuel_oils_residual-activity_combustion,Processed fuel oils - residual oil - combustion,kg-CO2e/t,3229.2,,BEIS,2021,GB,11/17/2021,Emission intensity for processed fuel oils (residual oil). The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
```
Comment (patstirling) [inline OpenEmissionFactorsDB.csv:1267]: suggest name "Processed fuel oils (resifual oil)" to be consistent with naming convention

File: OpenEmissionFactorsDB.csv [hunk 9]
```diff
@@ -1246,3 +1246,34 @@ Transport,Rail Travel,passenger_train-route_type_local-fuel_source_electricity,E
...
+Energy,Fuel,fuel_type_processed_fuel_oils_distillate-activity_combustion,Processed fuel oils - distillate oil - combustion,kg-CO2e/t,3230.28,,BEIS,2021,GB,11/17/2021,Emission intensity for processed fuel oils (distillate oil). The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
+Energy,Fuel,fuel_type_refinery_miscellaneous-activity_combustion,Refinery miscellaneous - combustion,kg-CO2e/t,2944.82,,BEIS,2021,GB,11/17/2021,Emission intensity for refinery miscellaneous. The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,fuel_combustion
```
Comment (patstirling) [inline OpenEmissionFactorsDB.csv:1269]: Suggest adding BEIS' comment to the description here to clarify: "Includes aromatic extracts, defoament solvents and other minor miscellaneous products"

**Review by patstirling — CHANGES_REQUESTED** [review]
- In all `IDs` lines _activity should be removed and _fuel_use should be added (see examples in comments)
- `Activity` field should be populated to match current fuel EFs (e.g. "combustion" not "fuel_combustion"
- `Name` field should no include activity (but do include use - see examples in comments)

File: OpenEmissionFactorsDB.csv (hunk 3 above)
Comment (greenercode) [reply OpenEmissionFactorsDB.csv:1279]: Screen Shot 2021-11-17 at 11 25 06 AM

File: OpenEmissionFactorsDB.csv [hunk 10]
```diff
@@ -1246,34 +1246,34 @@ Transport,Rail Travel,passenger_train-route_type_local-fuel_source_electricity,E
...
+Energy,Fuel,fuel_type_petroleum_coke-fuel_use_unknown,Petroleum coke ,kg-CO2e/t,3386.86,,BEIS,2021,GB,11/17/2021,Emission intensity for petroleum coke. The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,combustion
+Energy,Fuel,fuel_type_coal-fuel_use_home_produced_electricity_generation,Coal (electricity generation - home produced coal only),kg-CO2e/t,2248.82,,BEIS,2021,GB,11/17/2021,Emission intensity for coal (electricity generation - home produced coal only). The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,combustion
```
Comment (patstirling) [inline OpenEmissionFactorsDB.csv:1278]: Looked into this closer in BEIS - their naming is highly misleading! Their further guidance notes say:

"Coal used in power stations to generate electricity (only for coal produced in the UK)."

Suggest:
`id` = fuel_type_coal_produced_in_uk-fuel_use_electricity_generation
`name` = Coal produced in UK (electricity generation)
description = Emission intensity for UK-produced coal used in electricity generation

File: OpenEmissionFactorsDB.csv (hunk 1 above)
Comment (patstirling) [reply OpenEmissionFactorsDB.csv]: please transform all "unknown" -> "na" in the` ID` field

File: OpenEmissionFactorsDB.csv (hunk 3 above)
Comment (patstirling) [reply OpenEmissionFactorsDB.csv]: this will need "aviation spirit" removed

File: OpenEmissionFactorsDB.csv (hunk 2 above)
Comment (patstirling) [reply OpenEmissionFactorsDB.csv]: this should be fuel_use_aviation

File: OpenEmissionFactorsDB.csv (hunk 4 above)
Comment (patstirling) [reply OpenEmissionFactorsDB.csv]: this should be fuel_use_aviation

File: OpenEmissionFactorsDB.csv [hunk 11]
```diff
@@ -1246,34 +1246,34 @@ Transport,Rail Travel,passenger_train-route_type_local-fuel_source_electricity,E
...
+Energy,Fuel,fuel_type_lubricants-fuel_use_unknown,Lubricants ,kg-CO2e/t,3181.43,,BEIS,2021,GB,11/17/2021,Emission intensity for lubricants. The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,combustion
+Energy,Fuel,fuel_type_naphta-fuel_use_unknown,Naphtha ,kg-CO2e/t,3142.87,,BEIS,2021,GB,11/17/2021,Emission intensity for naphta. The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,combustion
```
Comment (patstirling) [inline OpenEmissionFactorsDB.csv:1264]: type here - "naphta" should be "naphtha" in 3 instances

File: OpenEmissionFactorsDB.csv [hunk 12]
```diff
@@ -1246,34 +1246,34 @@ Transport,Rail Travel,passenger_train-route_type_local-fuel_source_electricity,E
...
+Energy,Fuel,fuel_type_petrol_average_biofuel_blend-fuel_use_unknown,Petrol (average biofuel blend) ,kg-CO2e/t,2947.62,,BEIS,2021,GB,11/17/2021,Emission intensity for petrol (average biofuel blend). The fuels conversion factors should be used for primary fuel sources combusted at a site or in an asset owned or controlled by the reporting organisation. Available in BEIS Greenhouse Gas Reporting: Conversion Factors 2021.,https://www.gov.uk/government/publications/greenhouse-gas-reporting-conversion-factors-2021,combustion
```
Comment (patstirling) [inline OpenEmissionFactorsDB.csv:1265]: remove "average" from ID to align with comparable EFs in the category (e.g. line 398)

**Review by patstirling — CHANGES_REQUESTED** [review]

File: OpenEmissionFactorsDB.csv (hunk 9 above)
Comment (patstirling) [reply OpenEmissionFactorsDB.csv]: this yet to be updated

File: OpenEmissionFactorsDB.csv (hunk 3 above)
Comment (patstirling) [reply OpenEmissionFactorsDB.csv]: Spotted a rogue space on this line after "Natural gas" in the `name` field

File: OpenEmissionFactorsDB.csv (hunk 10 above)
Comment (patstirling) [reply OpenEmissionFactorsDB.csv]: `ID` field yet to be updated

**Review by patstirling — CHANGES_REQUESTED** [review]
Just a couple of issues to be resolved, see unresolved comments above

File: OpenEmissionFactorsDB.csv (hunk 3 above)
Comment (greenercode) [reply OpenEmissionFactorsDB.csv]: Which line exactly? I thought there is always supposed to be a space (before the comma separation) at the end of the name field?

Comment (patstirling) [reply OpenEmissionFactorsDB.csv]: Ah I see they are in all of the EFs now - no these will need to be removed

**Review by patstirling — APPROVED** [review]

**Review by patstirling — APPROVED** [review]

**Pull request merged**